certbot/certbot/_internal/storage.py

```diff
@@ -965,6 +975,44 @@ def ocsp_revoked(self, version: int) -> bool:
             logger.debug(str(e))
             return False
 
+    def get_renewalinfo(self, version:int, verify_ssl: bool, ua:str) -> Tuple[
+        bool, datetime.datetime, datetime.datetime]:
+        """from server in config try to get renewalinfo of certificate
+        if it sees error it will return datetime 1-01-01
+
+        :returns: tuple of [hasari, startdatetime, enddatetime]
+        :rtype: tuple[bool, datetime.datetime, datetime.datetime]
+        """
+        #if it's unittest it doesn't have any server
+        try:
+            serverurl = self.configuration["renewalparams"]["server"]
+        except KeyError:
+            serverurl = None
+
+        if serverurl is None:
+            return False, datetime.datetime(1,1,1), datetime.datetime(1,1,1)
+        try:
+            suffix = crypto_util.ariCertIdent(self.version('cert', version))
+            session = requests.session()
+            session.verify = verify_ssl
+            session.headers.update({'User-Agent': f'{ua}'})
+            r = session.get(serverurl, timeout = 1)
+            endpoint = r.json()["renewalInfo"]
+            r = session.get(f"{endpoint}/{suffix}", timeout = 1)
+            suggestedWindow = r.json()["suggestedWindow"]
+            start = parse_js_time(suggestedWindow['start'])
```

Comment:
```python

class RenewalInfo(josepy.JSONObjectWithFields):
    """
    4.2. RenewalInfo Objects
    https://www.ietf.org/archive/id/draft-ietf-acme-ari-07.html#name-renewalinfo-objects
    """

    class TimeWindow(josepy.JSONObjectWithFields):
        start: "datetime.datetime" = acme.messages.fields.RFC3339Field("start")
        end: "datetime.datetime" = acme.messages.fields.RFC3339Field("end")

    suggestedWindow: TimeWindow = josepy.Field("suggestedWindow")
    explanationURL: str = josepy.Field("explanationURL", omitempty=True)

Reply:
wonder why you want josepy class here: those are unsigned get so notthing to write JWS for.

Reply:
josepy is the ser/des available, it already provides rfc3399 parsing and can be used on unsigned data, minimizes code and allows consistency in message processing and error handling.

got to be 
```python
    suggestedWindow: TimeWindow = josepy.Field("suggestedWindow", decoder=TimeWindow.from_json)
```

though

```python
def test_RenewalInfo():
    from acme.messages import RenewalInfo
    import datetime
    data = {'suggestedWindow': dict(end='2025-03-01T08:53:18Z', start='2025-02-19T08:53:18Z')}
    ri = RenewalInfo.from_json(data)
    assert isinstance(ri.suggestedWindow.start, datetime.datetime)
```